### PR TITLE
remove users who are already following in share dropdown

### PR DESCRIPTION
**Pull request by ahsu1230**
and a fix for detecting youtube keep @2is10 
@andrewconner @stkem

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ angular.module('kifi')
         var useBigLayout = false;
         var strippedSchemeRe = /^https?:\/\//;
         var domainTrailingSlashRe = /^([^\/]*)\/$/;
-        var youtubeLinkRe = /^.*(youtu.be\/|v\/|e\/|u\/\w+\/|embed\/|v=)([^#\&\?]*).*/;
+        var youtubeVideoUrlRe = /^(?:https?:\/\/)?(?:youtu\.be|(?:www\.)?youtube(?:-nocookie)?\.com)\/(?:|user\/[^\/?#]+)?(?:|.*?[\/=])([a-zA-Z0-9_-]{11})\b/;
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js]: Thanks for the tip @2is10 
from https://github.com/kifi/keepit/pull/12517

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js]: Nice. No tests?

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js]: oh. yeaaaa that'd be a good idea

File: kifi-backend/modules/shoebox/angular/test/unit/common/util.spec.js [hunk 2]
```diff
@@ -37,4 +37,24 @@ describe('util', function () {
     });
   });
 
+  describe('util.isYoutube', function () {
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/test/unit/common/util.spec.js:40]: Thanks for the tips @yipingliao

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/test/unit/common/util.spec.js:40]: Awesome!

**Comment by andrewconner** [comment]
Replied to asana issue. This works, but also removes users from when the inviter types in their name. Check the issue for more info.

**Pull request merged**